drop/modules/aberrant-splicing-pipeline/fds_annotation.R

```diff
@@ -0,0 +1,67 @@
+### 20210604 klutz
+
+### basic annotations (start, end, none, both) for full fds
+createFDSAnnotations <- function(fds, txdb){
+  print("loading introns")
+  #seqlevelsStyle(fds) <- seqlevelsStyle(txdb)[1]
+  introns <- unique(unlist(intronsByTranscript(txdb)))
+  # reduce the introns to only the actually expressed introns
+  fds_known <- fds[unique(to(findOverlaps(introns, rowRanges(fds, type = "j"), type = "equal"))),]
+  grAnno <- rowRanges(fds_known, type="j")
+  anno_introns <- as.data.table(grAnno)
+  anno_introns <- anno_introns[,.(seqnames, start, end, strand)]
+  
+  #calculate extra columns with mean/median intron expression count
+  #add the new columns
+  print("adding median count to introns")
+  sampleCounts <- K(fds_known, type = "j")
+  anno_introns[, "meanCount" := rowMeans(sampleCounts)]
+  anno_introns[, "medianCount" := rowMedians(as.matrix(sampleCounts))]
+  
+  anno_introns_ranges <- makeGRangesFromDataFrame(anno_introns, keep.extra.columns = TRUE)
+  
+  ### get all fds junctions
+  fds_junctions <- rowRanges(fds, type = "j")
+  
+  ### Do the annotation just for the most used intron (highest median expression)
+  print("start calculating annotations")
+  annotations <- sapply(c(1:length(fds_junctions)), function(i){
```

Comment:
not sure how fast this is, but rather think of vectorized versions. granges is optimized to find overlaps between 1000s of ranges but not 1 by 1 over 1000s ranges.